core/src/consensus.rs

```diff
@@ -130,19 +130,57 @@ pub const HARD_FORK_INTERVAL: u64 = YEAR_HEIGHT / 2;
 /// Check whether the block version is valid at a given height, implements
 /// 6 months interval scheduled hard forks for the first 2 years.
 pub fn valid_header_version(height: u64, version: HeaderVersion) -> bool {
-	// uncomment below as we go from hard fork to hard fork
-	if height < HARD_FORK_INTERVAL {
-		version == HeaderVersion::default()
-	/* } else if height < 2 * HARD_FORK_INTERVAL {
-		version == 2
-	} else if height < 3 * HARD_FORK_INTERVAL {
-		version == 3
-	} else if height < 4 * HARD_FORK_INTERVAL {
-		version == 4
-	} else if height >= 5 * HARD_FORK_INTERVAL {
-		version > 4 */
-	} else {
-		false
+	let chain_type = global::CHAIN_TYPE.read().clone();
+	match chain_type {
+		global::ChainTypes::Mainnet => {
```

Comment:
This is duplicate code right? It is the same as the 3rd option of the `match`. Why not remove this one and just have floonet and "everything else"

Reply:
agreed with Jasper. if everything non-floonet is to behave the same then that should be reflected in a single branch of the code...

Reply:
:+1: On keeping `dev` and `mainnet` as similar as possible.